tests/fixtures/webserver.py

```diff
@@ -180,7 +180,7 @@ def requests(self) -> list[MockAPIServerRequest]:
 @pytest.fixture
 def mock_web_server() -> Generator[MockAPIServer, None, None]:
     """A test fixture that yields a usable mock web server for the lifetime of the test."""
-    _server = MockAPIServer("127.0.0.1", 10256)
+    _server = MockAPIServer("127.0.0.1", 0)
```

Comment:
This binds to an OS provided available random port. This was done so that this fixture can run in parallel, with multiple tests having this fixture running at once.